Applications/Utils/ModelPreparation/ComputeNodeAreasFromSurfaceMesh.cpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @brief Computes the areas associated nodes of the surface mesh.
+ *
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <fstream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "tclap/CmdLine.h"
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "BaseLib/FileTools.h"
+
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/readGeometryFromFile.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/MeshSurfaceExtraction.h"
+
+void writeToFile(std::string const& id_area_fname, std::string const& csv_fname,
+	std::vector<std::pair<std::size_t, double>> const& ids_and_areas,
+	std::vector<MeshLib::Node*> const& mesh_nodes)
+{
+	std::ofstream ids_and_area_out(id_area_fname);
+	if (!ids_and_area_out) {
+		ERR("Unable to open the file \"%s\" - aborting.", id_area_fname.c_str());
+		std::abort();
+	}
+	std::ofstream csv_out(csv_fname);
+	if (!csv_out) {
+		ERR("Unable to open the file \"%s\" - aborting.", csv_fname.c_str());
+		std::abort();
+	}
+
+	ids_and_area_out << std::setprecision(20);
+	csv_out << std::setprecision(20);
+
+	ids_and_area_out << ids_and_areas[0].first << " " << ids_and_areas[0].second;
+	csv_out << "ID x y z area node_id\n"; // CSV header
+	csv_out << 0 << " " << *(mesh_nodes[0])
+			<< ids_and_areas[0].second << " " << ids_and_areas[0].first;
```

Comment:
I do overlook something probably. Why is the output for id=0 not in the for-loop below? The loop starts from 1.